admin/controllers/service-manager.js

```diff
@@ -136,15 +137,40 @@ const controller = {
    * @param next
    * @returns {Promise.<void>}
    */
-  getUploadPupilCensus: async (req, res) => {
+  getUploadPupilCensus: async (req, res, next) => {
     res.locals.pageTitle = 'Upload pupil census'
     req.breadcrumbs(res.locals.pageTitle)
-
+    let pupilCensus
+    try {
+      pupilCensus = await pupilCensusService.getUploadedFile()
+    } catch (error) {
+      return next(error)
+    }
     res.render('service-manager/upload-pupil-census', {
-      breadcrumbs: req.breadcrumbs()
+      breadcrumbs: req.breadcrumbs(),
+      messages: res.locals.messages,
+      pupilCensus: pupilCensus
     })
   },
 
+  /**
+   * Submit pupil census.
+   * @param req
+   * @param res
+   * @param next
+   * @returns {Promise<*>}
+   */
+  postUploadPupilCensus: async (req, res, next) => {
+    const uploadFile = req.files && req.files.csvPupilCensusFile
```

Comment:
What is the upload file is not there?  Line 164 suggests there is the possibility of not having an upload file, but this caution is not evident in the `upload()` call.   Can it handle being called with an undefined variable?

Reply:
@jon-shipley this is now covered